src/pykka/_ref.py

```diff
@@ -70,9 +70,9 @@ def tell(self, message):
         """
         if not self.is_alive():
             raise ActorDeadError(f"{self} not found")
-        self.actor_inbox.put(Envelope(message))
+        self.actor_inbox.put(Envelope(message, delay=delay))
 
-    def ask(self, message, block=True, timeout=None):
+    def ask(self, message, block=True, timeout=None, delay=0):
```

Comment:
Document, type, units and semantics of delay in `ask`?

Reply:
Also should we allow this to be called with `None` and is that treated the same as zero? Or a negative delay? Or a duration from `datetime`?

Should we support an `at=None` API passing in a time?

Reply:
> Also should we allow this to be called with `None` and is that treated the same as zero? 

That's possible.

> Or a negative delay? 

That already works, although the resulting effect is that the message effectively gets a higher priority (because it will be put at the start of the `TimedInbox`). Not sure if that's a useful thing to allow or not.

> Or a duration from `datetime`?

Not sure what you mean here.

> Should we support an `at=None` API passing in a time?

That's kinda dangerous in view of the earlier discussion about `time.monotonic()`. We'll have to translate `at` into the monotonic time, and then the user adjusts the clock, or daylight savings happen, and the execution time will change.